components/org.wso2.carbon.identity.api.server.organization.user.invitation.management/org.wso2.carbon.identity.api.server.organization.user.invitation.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/user/invitation/management/v1/core/GuestApiServiceCore.java

```diff
@@ -199,6 +205,22 @@ public boolean deleteInvitation(String invitationId) {
      */
     public void acceptInvitation(AcceptanceRequestBody acceptanceRequestBody) {
 
+        InvitationCoreServiceImpl invitationCoreService = new InvitationCoreServiceImpl();
+        try {
+            invitationCoreService.acceptInvitation(acceptanceRequestBody.getConfirmationCode());
+        } catch (UserInvitationMgtException e) {
+            if (ERROR_CODE_INVALID_CONFIRMATION_CODE.getCode().equals(e.getErrorCode())) {
+                throw handleException(BAD_REQUEST, UserInvitationMgtConstants.ErrorMessage
+                        .ERROR_CODE_INVALID_CONFIRMATION_CODE_FOR_ACCEPTANCE, acceptanceRequestBody
+                        .getConfirmationCode());
+            } else if (ERROR_CODE_INVALID_USER.getCode().equals(e.getErrorCode())) {
+                throw handleException(BAD_REQUEST, UserInvitationMgtConstants.ErrorMessage
+                        .ERROR_CODE_INVALID_USER, StringUtils.EMPTY);
+            } else if (ERROR_CODE_USER_ALREADY_EXISTS.getCode().equals(e.getErrorCode())) {
+                throw handleException(BAD_REQUEST, UserInvitationMgtConstants.ErrorMessage
+                        .ERROR_CODE_EXISTING_USER, StringUtils.EMPTY);
+            }
```

Comment:
Don't we need the else part for default behaviour

Reply:
Added the default behavior